mucommander-core/src/main/java/com/mucommander/ui/action/impl/ToggleTerminalAction.java

```diff
@@ -82,8 +75,19 @@ public String getLabel() {
 
     @Override
     protected void toggleEnabledState() {
+        ensureIntegrationIsInitialized();
         AbstractFile currentFolder = mainFrame.getActivePanel().getCurrentFolder();
         setEnabled(currentFolder.getURL().getScheme().equals(LocalFile.SCHEMA));
     }
 
+    private void ensureIntegrationIsInitialized() {
+        if (terminalIntegration == null && mainFrame != null && mainFrame.getVerticalSplitPane() != null) {
```

Comment:
`mainFrame` cannot be null here because if it is null then we'll get NPE at line 79
and I'm not sure about the last part (`mainFrame.getVerticalSplitPane() != null`) - but maybe the right place to call `new TerminalIntegration(mainFrame, mainFrame.getVerticalSplitPane()` would be after initializing the vertical split pane rather than this class?

Reply:
I didn't want to glue vertical split pane to terminal functionality. It is just a container that can be used by other features in the future. I'm about to remove `mainFrame != null` - good point. PR will be updated soon.

> would be after initializing the vertical split pane rather than this class?
I can think about it this place is not good enough. Just let me know.

Reply:
> I can think about it if this place is not good enough. Just let me know.

I find it not intuitive to call it from toggleEnabledState, I'd prefer to call it explicitly from where we expect it the integration to be initialized but your explanation makes sense so let's get this in and continue to think about separately